### ISSUE TITLE
script.js

### ISSUE THREAD

**Issue by woowich1** [woowich1/asqwtoasdmasich#1]
jQuery(document).ready(function ($) {
    let theWheel;
    let wheelSpinning = false;

    function initWheel() {
        // Создаем колесо фортуны с заданными сегментами
        theWheel = new Winwheel({
            'canvasId': 'wheelCanvas',
            'numSegments': 6,
            'segments': [
                { 'fillStyle': 'linear-gradient(135deg, #4e73df, #1cc88a)', 'text': 'Скидка 1000 рублей' },
                { 'fillStyle': 'linear-gradient(135deg, #f6c23e, #e74a3b)', 'text': 'Фирменная кружка' },
                { 'fillStyle': 'linear-gradient(135deg, #4e73df, #1cc88a)', 'text': 'Скидка 500 рублей' },
                { 'fillStyle': 'linear-gradient(135deg, #f6c23e, #e74a3b)', 'text': 'Фирменный худи' },
                { 'fillStyle': 'linear-gradient(135deg, #4e73df, #1cc88a)', 'text': 'Сувенир от компании' },
                { 'fillStyle': 'linear-gradient(135deg, #f6c23e, #e74a3b)', 'text': 'Бесплатный урок' }
            ],
            'animation': {
                'type': 'spinToStop',
                'duration': 5,
                'spins': 8,
                'callbackFinished': showResult,
                'callbackSound': playTickSound,
                'soundTrigger': 'pin'
            },
            'pins': {
                'number': 6,
                'fillStyle': 'silver',
                'outerRadius': 4,
            },
            'lineWidth': 1,
            'textFontSize': 18,
            'textFillStyle': '#ffffff',
            'textAlignment': 'outer',
            'textMargin': 10,
            'textOrientation': 'curved',
            'textFontFamily': 'Arial',
        });

        $('#wheelCanvas').addClass('slow-spin');
    }

    function startWheelAnimation() {
        if (!wheelSpinning) {
            theWheel.stopAnimation(false);
            theWheel.rotationAngle = theWheel.rotationAngle % 360;
            theWheel.animation.spins = Math.floor(Math.random() * 3) + 5;
            theWheel.startAnimation();
            wheelSpinning = true;
        }
    }

    function showResult(indicatedSegment) {
        wheelSpinning = false;
        $('#wheelContainer').hide();
        $('#resultMessage').show();
        $('#prizeText').text('Поздравляем! Вы выиграли: ' + indicatedSegment.text);

        // Конфетти
        confetti.start();
        setTimeout(function () {
            confetti.stop();
        }, 5000);

        // Отправка данных с призом
        $.ajax({
            url: spinWheelAjax.ajax_url,
            type: 'POST',
            dataType: 'json',
            data: {
                action: 'spin_wheel_submit',
                name: $('#name').val().trim(),
                phone: $('#phone').val().trim(),
                prize: indicatedSegment.text,
                application_submitted: 'false'
            },
            success: function (response) {
                if (!response.success) {
                    $('#formErrors').html(response.data).show();
                }
            },
            error: function () {
                $('#formErrors').text('Произошла ошибка при отправке данных. Попробуйте еще раз.').show();
            }
        });
    }

    // Инициализация колеса при загрузке страницы
    $(window).on('load', function() {
        initWheel();
    });

    // Открытие модального окна при клике на мини колесо
    $('#openSpinWheelButton').click(function () {
        $('#spinWheelModal').modal('show');
        $(this).hide(); // Скрываем мини колесо после открытия модального окна
    });

    // Показываем мини колесо после закрытия модального окна
    $('#spinWheelModal').on('hidden.bs.modal', function () {
        $('#openSpinWheelButton').show();
    });

    // Обработка кликов на кнопки \"Да\" и \"Нет\"
    $('#hasLicenseButton').click(function () {
        $('#driverLicenseQuestion').hide();
        $('#licenseMessage').show();
        setTimeout(function () {
            $('#spinWheelModal').modal('hide');
        }, 5000);
    });

    $('#noLicenseButton').click(function () {
        $('#driverLicenseQuestion').hide();
        $('#leadFormContainer').show();
    });

    $('#leadForm').submit(function (e) {
        e.preventDefault();
        $('#formErrors').hide().text(''); // Очистка предыдущих ошибок

        let name = $('#name').val().trim();
        let phone = $('#phone').val().trim();
        let consent = $('#consent').is(':checked');

        let errors = [];

        if (name === '') {
            errors.push('Пожалуйста, введите ваше имя.');
        }

        if (phone === '') {
            errors.push('Пожалуйста, введите ваш номер телефона.');
        } else {
            let normalized_phone = phone.replace(/\D/g, '');
            if (normalized_phone.length < 10) {
                errors.push('Пожалуйста, введите корректный номер телефона.');
            }
        }

        if (!consent) {
            errors.push('Вы должны согласиться на обработку персональных данных.');
        }

        if (errors.length > 0) {
            $('#formErrors').html(errors.join('<br>')).show();
            return;
        }

        // Запуск анимации колеса после успешной проверки
        $('#leadFormContainer').hide();
        $('#wheelCanvas').removeClass('slow-spin');
        startWheelAnimation();
    });
});
